clusterloader2/pkg/measurement/common/resource_usage.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement/util/gatherers"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	resourceUsageMetricName = "ResourceUsageSummary"
+)
+
+func init() {
+	measurement.Register(resourceUsageMetricName, createResourceUsageMetricMeasurement)
+}
+
+func createResourceUsageMetricMeasurement() measurement.Measurement {
+	return &resourceUsageMetricMeasurement{}
+}
+
+type resourceUsageMetricMeasurement struct {
+	gatherer *gatherers.ContainerResourceGatherer
+}
+
+// Execute supports two actions:
+// - start - Stsrts resource metrics collecting.
+// - gather - Gathers and prints current resource usage metrics.
+func (e *resourceUsageMetricMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return summaries, err
+	}
+	provider, err := util.GetString(config.Params, "provider")
+	if err != nil {
+		return summaries, err
+	}
+	host, err := util.GetStringOrDefault(config.Params, "host", "")
+	if err != nil {
+		return summaries, err
+	}
+	masterOnly, err := util.GetBoolOrDefault(config.Params, "masterOnly", false)
```

Comment:
In the meantime we changed this one to enum (we added masterAndDNS).
Please reflect this here.

Reply:
Functions updated to newer version.